readme.txt

```diff
@@ -0,0 +1,39 @@
+=== Enable SVG Uploads ===
+Contributors: LewisCowles,hendridm
+Tags: svg,upload,media library,mime
+Requires at least: 4.0
```

Comment:
Ooh. I know there was a 4.x release that needed different settings. It's not BC compat tested, so I'm unsure if this will presently work with 4.0

Reply:
I will test it in 4.0 when I get a chance.

Reply:
TBH as a rule, only forward compat is really essential for me and my clients.

Reply:
True for me and my clients as well, but if you're talking about the wp.org repo, you may wish to have broader compatibility else get 1-star'd to oblivion. :stuck_out_tongue:

By the way, I just tested on WordPress 4.0.20 (which is the latest 4.0.x release) and it worked  with a minor change. As-is, I got the following error when I tried to upload to a post:

> PHP Strict standards: Only variables should be passed by reference in /app/public/wp-content/plugins/WordPressSVGPlugin/index.php on line 84

![WP 4.0.20 error](https://i.snag.gy/CQ8kY9.jpg "WP 4.0.20 error")

To fix, I had to change **Line 84** from:
```
$ext = strtolower( end( explode('.', $filename) ) );
```
to:
```
$ext = explode( '.', $filename );
$ext = strtolower( end( $ext ) );
```

...then it worked fine. This change is **not** required in WordPress 4.8.3 or 4.9-RC3, but I had to make the change to get it to work with WP 4.0.20 for some reason.

Reply:
Are they all running the same PHP version? I'm guessing it might be that `WP_DEBUG` is `true` for some reason, or the PHP is showing warnings from ini file or another plugin maybe.

TBF it is a dirty line, but I'm not sure the two line is the best fix. Thinking of that `svgz` suggestion, which should also be for `.svg.gz`.

Reply:
> Are they all running the same PHP version? I'm guessing it might be that WP_DEBUG is true for some reason ...

Yes, they were all running on the same VM, PHP 5.6. However, you were correct: `WP_DEBUG` was enabled (intentionally). When I disabled it, the error went away and everything worked fine.

Reply:
I'm pretty sure it should be fixed with the 1.8.3 update. I made a lib this AM to be a little more engineered about the extension using strings. It should work for 5.5+. The releases now have bundled deps separately to master branch.